lib/samson/hooks.rb

```diff
@@ -144,8 +144,10 @@ def fire(name, *args)
       end
 
       def render_views(name, view, *args)
-        hooks(name).each_with_object("".html_safe) do |partial, html|
-          html << view.render(partial, *args)
+        hooks(name).each_with_object("".html_safe) do |value, html|
+          partial, params_generator = value
+          new_args = params_generator ? [params_generator.call(*args)] : args
```

Comment:
make it render a unique template and then that template can render the
shared template

On Thu, Jun 23, 2016 at 10:34 AM, Zehua Liu <notifications@github.com>
wrote:

> In lib/samson/hooks.rb
> <https://github.com/zendesk/samson/pull/1058#discussion_r68278950>:
>
> > @@ -144,8 +144,10 @@ def fire(name, *args)
> >        end
> >
> >        def render_views(name, view, *args)
> > -        hooks(name).each_with_object("".html_safe) do |partial, html|
> > -          html << view.render(partial, *args)
> > +        hooks(name).each_with_object("".html_safe) do |value, html|
> > +          partial, params_generator = value
> > +          new_args = params_generator ? [params_generator.call(*args)] : args
>
> The same template is shared by two different plugins which have different
> ways of computing the template parameters. e.g., one determines whether to
> show the form based on whether there's flowdock configuration and the other
> decides based on slack configuration.
>
> Is there a different way to do it?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1058/files/1c0e3b4444e4a02023ed261f0c6443f9203994e5#r68278950>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAsZweczCs_vOKX_t42wIFKmNRHHnoVks5qOsONgaJpZM4Iyxjs>
> .
>